pynicotine/search.py

```diff
@@ -451,14 +452,69 @@ def create_search_result_list(self, searchterm, wordindex, excluded_words, parti
             log.add_debug("Error: DB closed during search, perhaps due to rescanning shares or closing the application")
             return None
 
-    def process_search_request(self, searchterm, username, token, direct=False):
+    def create_file_info_list(self, results, max_results, permission_level):
+        """ Given a list of file indices, retrieve the file information for each index """
+
+        reveal_buddy_shares = config.sections["transfers"]["reveal_buddy_shares"]
+        reveal_trusted_shares = config.sections["transfers"]["reveal_trusted_shares"]
+        is_buddy = (permission_level == "buddy")
+        is_trusted = (permission_level == "trusted")
+
+        fileinfos = []
+        private_fileinfos = []
+        num_fileinfos = 0
+
+        public_files = core.shares.share_dbs.get("files")
+        buddy_files = core.shares.share_dbs.get("buddyfiles")
+        trusted_files = core.shares.share_dbs.get("trustedfiles")
+
+        if public_files is None or buddy_files is None or trusted_files is None:
+            return num_fileinfos, fileinfos, private_fileinfos
+
+        for index in islice(results, min(len(results), max_results)):
+            file_path = core.shares.file_path_index[index]
+            fileinfo = public_files.get(file_path)
```

Comment:
```
fileinfo = public_files.get(repr(index))

if fileinfo is not None:
    file_path = core.shares.???```

Reply:
We no longer have a fileindex.db file with indices as keys, so the new code is correct. The indices used in wordindex.db go out of sync at some point when rescanning, however. I haven't figured out where yet.

Reply:
Simplified the index-related code in the scanner, now that I'm a bit more clear-headed. Hopefully it works better.

I'll add some error handling here in case the db files go out of sync, once I've verified that everything works well.